95.md

```diff
@@ -0,0 +1,88 @@
+NIP-95 - Storage and Shared File
+======
+`draft` `optional` `author:frbitten` 
+
+This NIP allows users to send binary data to be stored in the relay. This data can be small pieces of information or even large files.
+
+The intention is to combine the ease of NOSTR, with many clients being developed and a common communication protocol between all. To enable decentralized file sharing. NOSTR communication must work together with other protocols to get the best out of each technology. For lay users to be able to publish their photo, file, audio on several servers in a simple and easy way.
+
+Nostr event
+------------------
+This NIP specifies the use of the `1064` event type, having in `content` the binary data that you want to be stored in Base64 format.
+* `type` a string indicating the data type of the file. The MIME types format must be used (https://developer.mozilla.org/en-US/docs/Web/HTTP/Basics_of_HTTP/MIME_types/Common_types)
```

Comment:
If kind `1065` is optional, `type` is needed. What do you think?

Reply:
I think it's important to separate the metadadta (kind 1065) and the real data (kind 1064). To quote another part of this doc:

> Another defined event is the `1065` which is used as a header for the data contained in an event 1064. This way the data can be disclosed without overloading the communications when sending a large amount of data.

I don't need a "type" here if the "m" tag is present on 1065 -- it'll be redundant. And I'm not sure 1065 events should be considered optional. Clients should have a smaller 1065 metadata event available so that they can decide whether they want to download the (likely) larger 1064 event.  (Or, in the case of multi-part attachments, multiple events.)